📌 Module 1 – Customer Summary

For every customer, calculate:

Total Sales  
Total Profit  
Total Orders  
Total Quantity Purchased  
Average Order Value  

Expected Output

Customer	Sales	Profit	Orders	Qty	AOV

Business Insight  
Identify high-value customers.  
Measure customer profitability.  
Compare purchasing behavior.

In [0]:
%sql
select `Customer Name`, sum(`Sales`) as Total_Sales, sum(`Profit`) as Total_Profit, count(distinct `Order ID`) as Total_Orders, sum(`Quantity`)as Total_Quantity, 
round( sum(`Sales`) / COUNT(DISTINCT `Order ID`),2) as AOV, 
round(( sum(`Profit`) / sum(`Sales`))*100,2) as Profit_margin
from abc_retail_pvt_ltd.data_analytics.superstore
group by `Customer ID`, `Customer Name`
order by total_orders desc;

Customer Name,Total_Sales,Total_Profit,Total_Orders,Total_Quantity,AOV,Profit_margin
Emily Phan,5478.06079999999981600,144.95779999999988810,17,124,322.24,2.65
Eugene Hildebrand,3143.48300000000013740,117.81019999999995030,14,84,224.53,3.75
Suzanne McNair,5695.78999999999999100,610.11419999999993020,14,103,406.84,10.71
Bart Pistole,3350.87399999999997600,702.96419999999986580,14,85,239.35,20.98
Ben Ferrer,8430.14400000000019800,1980.32609999999996837,13,146,648.47,23.49
Chloris Kastensmidt,3154.85500000000001450,141.28309999999996320,13,122,242.68,4.48
Noel Staavos,2964.82299999999995030,-234.77140000000007870,13,113,228.06,-7.92
Zuschuss Carroll,8025.70699999999987305,-1032.14900000000082210,13,105,617.36,-12.86
Xylona Preis,2503.15599999999998700,675.95839999999996130,13,121,192.55,27.00
Erin Ashbrook,2846.70499999999999600,-52.74060000000010540,13,87,218.98,-1.85


📌 Module 2 – Customer Ranking

Rank customers based on:

Revenue
Profit
Number of Orders

Business Insight

Find:

Highest spending customers  
Most profitable customers  
Most frequent customers

In [0]:
%sql
select `Customer Name`, `Customer ID`, round(Sum(`Sales`),2) as Total_Sales, round(Sum(Profit),2) as Total_Profit, Count(Distinct `Order ID`) as Num_Orders,
rank() over( order by Sum(`Sales`) desc) as Revenue_Rank,
rank() over(order by Sum(Profit) desc) as Profit_Rank,
rank() over(order by Count(Distinct `Order ID`) desc) as Orders_Rank
from abc_retail_pvt_ltd.data_analytics.superstore
group by `Customer ID`, `Customer Name`;

Customer Name,Customer ID,Total_Sales,Total_Profit,Num_Orders,Revenue_Rank,Profit_Rank,Orders_Rank
Sean Miller,SM-20320,25043.05,-1980.74,5,1,797,472
Tamara Chand,TC-20980,19052.22,8981.32,5,2,1,472
Raymond Buch,RB-19360,15117.34,6976.10,6,3,2,364
Tom Ashbrook,TA-21385,14595.62,4703.79,4,4,6,607
Adrian Barton,AB-10105,14473.57,5444.81,10,5,5,61
Ken Lonsdale,KL-16645,14175.23,806.85,12,6,126,15
Sanjit Chand,SC-20095,14142.33,5757.41,9,7,3,98
Hunter Lopez,HL-15040,12873.30,5622.43,6,8,4,364
Sanjit Engle,SE-20110,12209.44,2650.68,11,9,13,35
Christopher Conant,CC-12370,12129.07,2177.05,5,10,19,472


📌 Module 3 – Customer Segmentation

This is used in almost every company.

Create segments like:

Sales	     Segment
>10000  	 Platinum
5000–10000	 Gold
1000–5000	 Silver
<1000	     Bronze

We'll use a CASE WHEN statement.

In [0]:
%sql
select `Customer ID`, `Customer Name`, Round(sum(`Sales`),2) as Total_Sales,
CASE 
when sum(`Sales`) > 10000 then 'Platinum'
when sum(`Sales`) between 5000 and 10000 then 'Gold'
when sum(`Sales`) between 2000 and 5000 then 'Silver'
else 'Bronze'
end as Segment
from abc_retail_pvt_ltd.data_analytics.superstore
group by `Customer ID`, `Customer Name`
order by Total_Sales desc;

Customer ID,Customer Name,Total_Sales,Segment
SM-20320,Sean Miller,25043.05,Platinum
TC-20980,Tamara Chand,19052.22,Platinum
RB-19360,Raymond Buch,15117.34,Platinum
TA-21385,Tom Ashbrook,14595.62,Platinum
AB-10105,Adrian Barton,14473.57,Platinum
KL-16645,Ken Lonsdale,14175.23,Platinum
SC-20095,Sanjit Chand,14142.33,Platinum
HL-15040,Hunter Lopez,12873.30,Platinum
SE-20110,Sanjit Engle,12209.44,Platinum
CC-12370,Christopher Conant,12129.07,Platinum


📌 Module 4 – Customer Profitability

Business asks:

Which customers generate high sales but low profit?

These customers may be receiving excessive discounts.

Find them.

What should you look for?

Customers having

High Sales  
Low Profit Margin

These customers deserve investigation.

In [0]:
%sql
select `Customer Name`, `Customer ID` , round(sum(`Sales`),2) as Total_Sales, round(sum(`Profit`),2) as Total_Profit, round((sum(`Profit`)/sum(`Sales`))*100,2) as Profit_Margin, 
rank() over( Order by sum(`Sales`) desc) as Sales_Rank
from abc_retail_pvt_ltd.data_analytics.superstore
group by `Customer ID`, `Customer Name`
order by sales_rank

Customer Name,Customer ID,Total_Sales,Total_Profit,Profit_Margin,Sales_Rank
Sean Miller,SM-20320,25043.05,-1980.74,-7.91,1
Tamara Chand,TC-20980,19052.22,8981.32,47.14,2
Raymond Buch,RB-19360,15117.34,6976.10,46.15,3
Tom Ashbrook,TA-21385,14595.62,4703.79,32.23,4
Adrian Barton,AB-10105,14473.57,5444.81,37.62,5
Ken Lonsdale,KL-16645,14175.23,806.85,5.69,6
Sanjit Chand,SC-20095,14142.33,5757.41,40.71,7
Hunter Lopez,HL-15040,12873.30,5622.43,43.68,8
Sanjit Engle,SE-20110,12209.44,2650.68,21.71,9
Christopher Conant,CC-12370,12129.07,2177.05,17.95,10


📌 Module 5 – Top 20 Customers

Business asks:

If we lose our top 20 customers, how much revenue do we lose?

Find:  

Top 20 customers  
Their revenue  
Contribution to total sales

In [0]:
%sql
with TopCustomersRevenue as
(select `Customer ID`, `Customer Name`, round(sum(`Sales`), 2) as Total_Revenue, 
rank() over(order by sum(`Sales`) desc) as Customer_Rank
from abc_retail_pvt_ltd.data_analytics.superstore
group by `Customer ID`, `Customer Name`
)

select * from TopCustomersRevenue
where Customer_Rank <= 20;

Customer ID,Customer Name,Total_Revenue,Customer_Rank
SM-20320,Sean Miller,25043.05,1
TC-20980,Tamara Chand,19052.22,2
RB-19360,Raymond Buch,15117.34,3
TA-21385,Tom Ashbrook,14595.62,4
AB-10105,Adrian Barton,14473.57,5
KL-16645,Ken Lonsdale,14175.23,6
SC-20095,Sanjit Chand,14142.33,7
HL-15040,Hunter Lopez,12873.30,8
SE-20110,Sanjit Engle,12209.44,9
CC-12370,Christopher Conant,12129.07,10


In [0]:
%sql
with TopCustomersRevenue as
(
select `Customer ID`, `Customer Name`, round(sum(`Sales`), 2) as Total_Revenue, 
rank() over(order by sum(`Sales`) desc) as Customer_Rank
from abc_retail_pvt_ltd.data_analytics.superstore
group by `Customer ID`, `Customer Name`
)

select sum(Total_Revenue) from TopCustomersRevenue
where Customer_Rank <= 20;

sum(Total_Revenue)
265060.25


In [0]:
%sql
-- Total Revenue
select round(sum(`Sales`), 2) as Total_Revenue
from abc_retail_pvt_ltd.data_analytics.superstore

Total_Revenue
2326534.35


In [0]:
%sql
-- Contribution of top 20 customers
WITH CustomerRanking AS
(
SELECT `Customer ID`, `Customer Name`, SUM(`Sales`) AS Total_Sales,
RANK() OVER(ORDER BY SUM(`Sales`) DESC) AS Customer_Rank
FROM abc_retail_pvt_ltd.data_analytics.superstore
GROUP BY `Customer ID`, `Customer Name`)

SELECT ROUND((SUM(Total_Sales)/(SELECT SUM(Sales) FROM abc_retail_pvt_ltd.data_analytics.superstore))*100,2) AS Revenue_Contribution_Percentage
FROM CustomerRanking
WHERE Customer_Rank<=20;

Revenue_Contribution_Percentage
11.39


Finding

The top 20 customers contribute 11.39% of the company's total revenue.

Business Insight

A relatively small group of customers generates a significant portion of total sales, indicating that these customers are strategically important.

Recommendation

Implement customer retention initiatives such as loyalty programs, personalized offers, proactive account management, and regular engagement to minimize churn among these high-value customers.

Develop a customer retention strategy through personalized communication, loyalty programs, priority customer support, and exclusive benefits to reduce churn among high-value customers.

Recommendation
Introduce a customer loyalty program for top customers.  
Provide personalized promotions and exclusive offers based on purchase history.  
Maintain regular engagement through dedicated account management and targeted communication.  
Monitor purchasing trends to identify early signs of customer churn.